silkworm/silkrpc/commands/eth_api.cpp

```diff
@@ -2015,6 +2016,64 @@ awaitable<void> EthereumRpcApi::handle_eth_unsubscribe(const nlohmann::json& req
     co_return;
 }
 
+// https://eth.wiki/json-rpc/API#eth_feehistory
+awaitable<void> EthereumRpcApi::handle_fee_history(const nlohmann::json& request, nlohmann::json& reply) {
+    const auto& params = request["params"];
+    if (params.size() != 3) {
+        const auto error_msg = "invalid eth_feeHistory params: " + params.dump();
+        SILKRPC_ERROR << error_msg << "\n";
+        reply = make_json_error(request["id"], 100, error_msg);
+        co_return;
+    }
+
+    uint64_t block_count;
+    if (params[0].is_string()) {
+        const auto value = params[0].get<std::string>();
+        block_count = std::stoul(value, nullptr, 16);
+    } else {
+        block_count = params[0].get<uint64_t>();
+    }
+    const auto newest_block = params[1].get<std::string>();
+    const auto reward_percentile = params[2].get<std::vector<int32_t>>();
+
+    SILKRPC_LOG << "block_count: " << block_count
+                << ", newest_block: " << newest_block
+                << ", reward_percentile size: " << reward_percentile.size() << "\n";
+
+    auto tx = co_await database_->begin();
+
+    try {
+        ethdb::TransactionDatabase tx_database{*tx};
+
+        rpc::fee_history::BlockProvider block_provider = [this, &tx_database](uint64_t block_number) {
+            return core::read_block_by_number(*(this->block_cache_), tx_database, block_number);
+        };
+        rpc::fee_history::ReceiptsProvider receipts_provider = [&tx_database](const BlockWithHash& block_with_hash) {
+            return core::get_receipts(tx_database, block_with_hash);
+        };
+
+        rpc::fee_history::FeeHistoryOracle oracle{block_provider, receipts_provider};
+
+        /*const auto block_number = co_await core::get_block_number(newest_block, tx_database);*/
+        auto fee_history = co_await oracle.fee_history(4417196, block_count, reward_percentile);
```

Comment:
I guess that hard-coded block number is a reminder of some test... probably the commented line above is correct

Reply:
You're right